From 5a01d403488053fd423961c4cea3fbdc55fe5f3e Mon Sep 17 00:00:00 2001
From: Christian Dupuis <cd@docker.com>
Date: Tue, 10 Feb 2026 01:07:45 +0100
Subject: [PATCH] Bind string as string_view via temporary variable

Cast string to string_view using a temporary variable to allow pass-by-reference to prevent error: cannot bind non-const lvalue reference of type 'std::basic_string_view&' to an rvalue of type 'std::__cxx11::basic_string::__sv_type' {aka 'std::basic_string_view'}
---
 src/node_sea.cc | 19 ++++++++++++-------
 1 file changed, 12 insertions(+), 7 deletions(-)

diff --git a/src/node_sea.cc b/src/node_sea.cc
index bffdc72d1d1..ed96c45b617 100644
--- a/src/node_sea.cc
+++ b/src/node_sea.cc
@@ -375,29 +375,34 @@ std::optional<SeaConfig> ParseSingleExecutableConfig(
       return std::nullopt;
     }
     if (key == "main") {
-      if (field.value().get_string().get(result.main_path) ||
-          result.main_path.empty()) {
+       std::string_view main_path_view = result.main_path;
+       if (field.value().get_string().get(main_path_view) ||
+           main_path_view.empty()) {
         FPrintF(stderr,
                 "\"main\" field of %s is not a non-empty string\n",
                 config_path);
         return std::nullopt;
       }
+      result.main_path = main_path_view;
     } else if (key == "output") {
-      if (field.value().get_string().get(result.output_path) ||
-          result.output_path.empty()) {
-        FPrintF(stderr,
+      std::string_view output_path_view = result.output_path;
+      if (field.value().get_string().get(output_path_view) ||
+          output_path_view.empty()) {        FPrintF(stderr,
                 "\"output\" field of %s is not a non-empty string\n",
                 config_path);
         return std::nullopt;
       }
+      result.output_path = output_path_view;
     } else if (key == "executable") {
-      if (field.value().get_string().get(result.executable_path) ||
-          result.executable_path.empty()) {
+      std::string_view executable_path_view = result.executable_path;
+      if (field.value().get_string().get(executable_path_view) ||
+          executable_path_view.empty()) {
         FPrintF(stderr,
                 "\"executable\" field of %s is not a non-empty string\n",
                 config_path);
         return std::nullopt;
       }
+      result.executable_path = executable_path_view;
     } else if (key == "disableExperimentalSEAWarning") {
       bool disable_experimental_sea_warning;
       if (field.value().get_bool().get(disable_experimental_sea_warning)) {
-- 
2.52.0
